jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -0,0 +1,291 @@
+        LayerType.RESOURCES, resourceFilesDirectory, path -> true, RESOURCES_PATH);
+    if (!classpath.contains(RESOURCES_PATH.toString())) {
+      classpath.add(RESOURCES_PATH.toString());
+    }
+    return this;
+  }
+
+  /**
+   * Adds classes to /app/classes on the image.
+   *
+   * @param classFilesDirectory the directory containing the class files
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addClasses(Path classFilesDirectory) throws IOException {
+    if (!Files.exists(classFilesDirectory) || !Files.isDirectory(classFilesDirectory)) {
+      throw new IOException(
+          "Adding classes failed: '" + classFilesDirectory + "' is not a directory");
+    }
+    layerConfigurationsBuilder.addDirectoryContents(
+        LayerType.CLASSES, classFilesDirectory, path -> true, CLASSES_PATH);
+    if (!classpath.contains(CLASSES_PATH.toString())) {
+      classpath.add(CLASSES_PATH.toString());
+    }
+    return this;
+  }
+
+  /**
+   * Adds additional files to the image's classpath.
+   *
+   * @param otherFiles the list of files to add. Files are added to /app/other on the container
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addToClasspath(List<Path> otherFiles) throws IOException {
+    for (Path file : otherFiles) {
+      if (!Files.exists(file)) {
+        continue;
+      }
+      if (Files.isDirectory(file)) {
+        layerConfigurationsBuilder.addDirectoryContents(
+            LayerType.EXTRA_FILES, file, path -> true, OTHERS_PATH);
+      } else {
+        layerConfigurationsBuilder.addFile(
+            LayerType.EXTRA_FILES, file, OTHERS_PATH.resolve(file.getFileName()));
+      }
+    }
+    if (!classpath.contains(OTHERS_PATH.toString())) {
+      classpath.add(OTHERS_PATH.toString());
+    }
+    return this;
+  }
+
+  /**
+   * Adds additional files to the image's classpath.
+   *
+   * @param otherFiles the list of files to add. Files are added to /app/other on the container
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addToClasspath(Path... otherFiles) throws IOException {
+    return addToClasspath(Arrays.asList(otherFiles));
+  }
+
+  /**
+   * Sets the JVM flags to use when starting the application.
+   *
+   * @param jvmFlags the list of JVM flags
+   * @return this
+   */
+  public JavaContainerBuilder setJvmFlags(List<String> jvmFlags) {
+    this.jvmFlags = ImmutableList.copyOf(jvmFlags);
+    return this;
+  }
+
+  /**
+   * Sets the JVM flags to use when starting the application.
+   *
+   * @param jvmFlags the list of JVM flags
+   * @return this
+   */
+  public JavaContainerBuilder setJvmFlags(String... jvmFlags) {
+    this.jvmFlags = ImmutableList.copyOf(jvmFlags);
+    return this;
+  }
+
+  /**
+   * Sets the main class used to start the application on the image.
```

Comment:
I think we can expose `MainClassFinder` to users of Jib Core and point them to use that if they want to have an automatic way of finding the main class among a set of class files.